File: academic-system-java/src/main/java/org/example/academic/system/repository/PersistenceStrategy.java
package org.example.academic.system.repository;

import org.example.academic.system.model.AcademicClass;
import java.util.List;

public interface PersistenceStrategy {
    void save(List<AcademicClass> classes);
    List<AcademicClass> load(); // ADICIONE ESTA LINHA
    String getFormatName();
}

File: academic-system-python/academic_system/repository/persistence_strategy.py
from abc import ABC, abstractmethod
from typing import List

from ..model.academic_class import AcademicClass


class PersistenceStrategy(ABC):
    @abstractmethod
    def save(self, classes: List[AcademicClass]) -> None:
        pass

    @abstractmethod
    def load(self) -> List[AcademicClass]:
        pass

    @abstractmethod
    def get_format_name(self) -> str:
        pass
